tests/foreman/api/test_convert2rhel.py

```diff
@@ -0,0 +1,324 @@
+        rh_repo = entities.Repository(id=rh_repo_id).read()
+        rh_repo.sync(timeout=2500)
+        rh_repos.append(rh_repo)
+    return rh_repos
+
+
+@pytest.fixture(scope='module')
+def enable_rhel7_subscriptions(module_org, manifest):
+    """Enable and sync RHEL7 server rmps repo"""
+    return enable_rhel_repos(module_org, ['rhel7'])
+
+
+@pytest.fixture(scope='module')
+def enable_rhel8_subscriptions(module_org, manifest):
+    """Enable and sync RHEL8 baseos and appstream repos"""
+    return enable_rhel_repos(module_org, ['rhel8_bos', 'rhel8_aps'])
+
+
+@pytest.fixture
+def centos_host(
+    request, default_sat, module_org, smart_proxy_location, module_promoted_cv, module_lce, version
+):
+    """Deploy and register Centos host"""
+    rh_repos = request.getfixturevalue(f"enable_rhel{version}_subscriptions")
+    conf = {
+        "workflow": "deploy-centos",
+        "deploy_scenario": "centos",
+        "deploy_rhel_version": version,
+    }
+    with VMBroker(**conf, host_classes={'host': ContentHost}) as host:
+        host.install_katello_ca(default_sat)
+        # updating centos packages on CentOS 8 is necessary for conversion
+        if version == '8':
+            host.execute("yum update -y centos-*")
+        repo = create_repo(module_org, CONVERT_TO_RHEL_REPO.format(version))
+        cv = update_cv(module_promoted_cv, module_lce, rh_repos + [repo])
+        c2r_sub = entities.Subscription(organization=module_org, name=repo.product.name).search()[0]
+        act_key = create_activation_key(module_org, module_lce, cv, c2r_sub.id)
+        register_host(act_key, module_org, smart_proxy_location, host.hostname)
+        yield host
+        # close ssh session before teardown, because of reboot in conversion it may cause problems
+        host.close()
+
+
+@pytest.fixture
+def oracle_host(
+    request,
+    default_sat,
+    module_org,
+    smart_proxy_location,
+    module_promoted_cv,
+    module_lce,
+    ssl_cert,
+    version,
+):
+    """Deploy and register Oracle host"""
+    rh_repos = request.getfixturevalue(f"enable_rhel{version}_subscriptions")
+    conf = {
+        "workflow": "deploy-oracle-linux",
+        "deploy_scenario": "oracle",
+        "deploy_rhel_version": version,
+    }
+    with VMBroker(**conf, host_classes={'host': ContentHost}) as host:
+        # disable rhn-client-tools because it obsoletes the subscribtion manager package
+        host.execute('echo "exclude=rhn-client-tools" >> /etc/yum.conf')
+        # install and set correct kernel, based on convert2rhel docs
+        result = host.execute(
+            'yum install -y kernel && '
+            'grubby --set-default /boot/vmlinuz-'
+            '`rpm -q --qf "%{BUILDTIME}\t%{EVR}.%{ARCH}\n" kernel | sort -nr | head -1 | cut -f2`'
+        )
+        assert result.status == 0
+        host.install_katello_ca(default_sat)
+        try:
+            result = host.execute('reboot', timeout='5s')
```

Comment:
Do you think https://github.com/SatelliteQE/robottelo/blob/67c8165fa670e36584fdd9f06ffb0af5aea51b80/robottelo/hosts.py#L193 could be a replacement here?

Reply:
:+1: Didn't know about this option, I will try it